be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -196,6 +206,17 @@ Status VerticalGeneralTabletWriter::flush_columns() {
     DCHECK(_segment_writers[_current_writer_index]);
     RETURN_IF_ERROR(flush_columns(&_segment_writers[_current_writer_index]));
     _current_writer_index = 0;
+
+    if (_segment_writer_finalize_token != nullptr) {
+        // Wait all async tasks finished
+        for (auto& future : _futures) {
+            if (auto st = future.get(); !st.ok()) {
+                LOG(WARNING) << "Fail to finalize segment, " << st;
+                return st;
+            }
+        }
+        _futures.clear();
+    }
```

Comment:
```suggestion
    Statis ret;
    if (_segment_writer_finalize_token != nullptr) {
        // Wait all async tasks finished
        for (auto& future : _futures) {
            if (auto st = future.get(); !st.ok()) {
                LOG(WARNING) << "Fail to finalize segment: " << st;
                ret.update(st);
            }
        }
        _futures.clear();
    }
    return ret;
```

Reply:
I think the function can return directly when any error encountered ?